src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -66,10 +68,15 @@ int message_new_empty(sd_netlink *rtnl, sd_netlink_message **ret) {
 int message_new(sd_netlink *rtnl, sd_netlink_message **ret, uint16_t type) {
         _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *m = NULL;
         const NLType *nl_type;
+        const NLTypeSystem *type_system_root;
         size_t size;
         int r;
 
-        r = type_system_get_type(&type_system_root, &nl_type, type);
+        type_system_get_root(rtnl->protocol, &type_system_root);
+        if (type_system_root == NULL)
```

Comment:
our coding style suggests not to compare ptrs with `== NULL` and `!= NULL`, but simply rely on C's degrade-to-bool feature, i.e. `if (p == NULL)` → `if (p)`. Also see CODING_STYLE

Reply:
also, this can't actually fail, can it?